qutebrowser/browser/webkit/webkithistory.py

```diff
@@ -48,7 +50,8 @@ def historyContains(self, url_string):
         Return:
             True if the url is in the history, False otherwise.
         """
-        return url_string in self._history.history_dict
+        with debug.log_time('sql', 'historyContains'):
+            return url_string in self._history
```

Comment:
That won't work, as we can't cache `False` - a URL which was previously not visited can be visited somewhen in the future. We can only cache `True`.

And nope, it does not. I've even seen it call this for the same URL twice in a row...